opengauss jsonb type error

### Description

2023-07-06 11:27:07.582 - Can't read column 'geom_json' value
org.jkiss.dbeaver.model.exec.DBCException: Error converting string to composite type
	at org.jkiss.dbeaver.ext.postgresql.model.data.PostgreStructValueHandler.getValueFromObject(PostgreStructValueHandler.java:110)
	at org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCComplexValueHandler.fetchColumnValue(JDBCComplexValueHandler.java:50)
	at org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCAbstractValueHandler.fetchValueObject(JDBCAbstractValueHandler.java:49)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetDataReceiver.fetchRow(ResultSetDataReceiver.java:126)
	at org.jkiss.dbeaver.model.impl.jdbc.struct.JDBCTable.readData(JDBCTable.java:201)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:189)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5101)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.jkiss.dbeaver.DBException: Composite type 'jsonb' has no attributes
	at org.jkiss.dbeaver.ext.postgresql.model.data.PostgreStructValueHandler.convertStringToStruct(PostgreStructValueHandler.java:184)
	at org.jkiss.dbeaver.ext.postgresql.model.data.PostgreStructValueHandler.getValueFromObject(PostgreStructValueHandler.java:107)
	... 10 more

### DBeaver Version

dbeaver-ce-23.1.1

### Operating System

windows10

### Database and driver

opengauss 3.0.0

### Steps to reproduce

_No response_

### Additional context

_No response_

--- a/plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/data/PostgreValueHandlerProvider.java
+++ b/plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/data/PostgreValueHandlerProvider.java
@@ -29,6 +29,7 @@ import org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCContentValueHandler;
 import org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCNumberValueHandler;
 import org.jkiss.dbeaver.model.impl.jdbc.data.handlers.JDBCStandardValueHandlerProvider;
 import org.jkiss.dbeaver.model.struct.DBSTypedObject;
+import org.jkiss.utils.CommonUtils;
 
 import java.sql.Types;
 
@@ -45,10 +46,17 @@ public class PostgreValueHandlerProvider extends JDBCStandardValueHandlerProvide
 //            return PostgreEnumValueHandler.INSTANCE;
 //        }
         int typeID = typedObject.getTypeID();
+        String typeName = typedObject.getTypeName();
         switch (typeID) {
             case Types.ARRAY:
                 return PostgreArrayValueHandler.INSTANCE;
             case Types.STRUCT:
+                if (CommonUtils.isNotEmpty(typeName)
+                    && (PostgreConstants.TYPE_JSONB.equals(typeName) || PostgreConstants.TYPE_JSON.equals(typeName))
+                ) {
+                    // The special case for the OpenGauss database, which returns json types as a struct type.
+                    return PostgreJSONValueHandler.INSTANCE;
+                }
                 return PostgreStructValueHandler.INSTANCE;
             case Types.DATE:
             case Types.TIME:
@@ -61,7 +69,7 @@ public class PostgreValueHandlerProvider extends JDBCStandardValueHandlerProvide
                     return new PostgreDateTimeValueHandler(preferences);
                 }
             default:
-                switch (typedObject.getTypeName()) {
+                switch (typeName) {
                     case PostgreConstants.TYPE_VARBYTE:
                         return JDBCContentValueHandler.INSTANCE;
                     case PostgreConstants.TYPE_JSONB:
@@ -85,7 +93,7 @@ public class PostgreValueHandlerProvider extends JDBCStandardValueHandlerProvide
                     case PostgreConstants.TYPE_INTERVAL:
                         return PostgreIntervalValueHandler.INSTANCE;
                     default:
-                        if (PostgreConstants.SERIAL_TYPES.containsKey(typedObject.getTypeName())) {
+                        if (PostgreConstants.SERIAL_TYPES.containsKey(typeName)) {
                             return new JDBCNumberValueHandler(typedObject, preferences);
                         }
                         if (typeID == Types.OTHER || typedObject.getDataKind() == DBPDataKind.STRING) {

